docs/architecture/adr/0023-integration-identifiers.md

```diff
@@ -0,0 +1,89 @@
+
+- **Maintain current integration method** - Make no changes to validation or registration of traffic
+  and expect self-regulation of integrations.
+- **Support only Bitwarden integrations** - Utilize application signature checks and other security
+  features to only allow Bitwarden-developed clients to connect to server-side components, therefore
+  constraining traffic to known entities.
+- **Validate baseline expectations** - Inspect and eventually enforce several checks on traffic so
+  that all integrations provide a minimum amount of information on their state so that the platform
+  can be better monitored.
+- **Register integrations along with validation** - Develop a simple registration method for
+  integrations so that they are enumerated and known to Bitwarden, whether internally or externally
+  developed. Associate validation with registered integrations for enhanced developer feedback.
+
+## Decision Outcome
+
+Chosen option: **Register integrations along with validation**.
+
+While this is more effort for Bitwarden to develop than just adding validation, combining efforts to
+expect more from integrations' request data with also providing a registration identifier is a
+simple addition.
+
+### Positive Consequences
+
+- Integrations are registered and known to the platform for sensible traffic identification.
+- Bitwarden's operations group receives data to aid in maintainability of the platform, especially
+  to keep things running well for users while keeping out bad actors.
+- Feedback to Bitwarden support teams on version usage is available.
+- Bitwarden's support policy of a certain number of major versions can be more actively enforced.
+
+### Negative Consequences
+
+- Integrations will need to make the effort to register with Bitwarden and adjust their requests,
+  potentially with brief disruptions for them.
+- Very small latency may be added for the additional validations.
+
+### Plan
+
+Documentation will be provided on the Help Center or this contributing docs site indicating the
+minimally-required request headers for all clients to provide when communicating to the Bitwarden
+platform. Release notes will include mention of this and the future enforcement after three major
+releases. Documentation will also be expanded to provide guidance on how unofficial clients should
+form their provided client version specifically to accurately represent supported "windows" of
+client-server interactivity.
+
+Operations teams will perform the necessary development to validate that required headers are
+present and enable its enforcement after that time; requests will be rejected as a `400 Bad Request`
+when required headers are missing and a `403 Forbidden` when provided headers or their values are
+not supported.
+
+A process will be established for integrations to submit support tickets requesting a client
+identifier and secret. Customer Success will work with operations teams to register integrations and
```

Comment:
I don't see how a secret could be used that is actually kept secret. Is there a clearer plan here?

Reply:
Eliminated in a2e7124771e12ec69748c00ed5bcc00883c3d7a6.